### PR TITLE
Murmur2 hashes of byte strings with a length that is a factor of 4

### PR DESCRIPTION
**Pull request by johanandren**
Refs #30696

Credits for finding and suggesting fix goes to @coertel-yw

### CODE REVIEW

File: akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/internal/Murmur2Spec.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.nio.charset.StandardCharsets.UTF_8
+
+class Murmur2Spec extends AnyWordSpecLike with Matchers {
+  "The Murmur2 hashing" must {
+    // expected correct hash values from the kafka murmur2 impl
+    // https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L500
```
Comment (andreaTP) [inline akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/internal/Murmur2Spec.scala:15]:
```suggestion
    // https://github.com/apache/kafka/blob/db42afd6e24ef4291390b4d1c1f10758beedefed/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L500
```

use a permalink

**Comment by andreaTP** [comment]
accidentally committed on top but should be good @johanandren

**Review by andreaTP — APPROVED** [review]

**Comment by coertel-yw** [comment]
@johanandren thank you for taking charge! my apologies for slacking off after the initial commits ...

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**